tools/build_langserver/langserver/fs.go

```diff
@@ -0,0 +1,74 @@
+package langserver
+
+import (
+	"context"
+	"encoding/json"
+
+	"tools/build_langserver/lsp"
+
+	"github.com/sourcegraph/jsonrpc2"
+)
+
+func (h *LsHandler) handleFSRequests(ctx context.Context, req *jsonrpc2.Request) (result interface{}, err error) {
+	if !isFileSystemMethod(req) {
+		return nil, nil
+	}
+
+	log.Info("Handling fs method %s, with param %s", req.Method, req.Params)
+
+	switch req.Method {
+	case "textDocument/didOpen":
+		var params lsp.DidOpenTextDocumentParams
+		if err := json.Unmarshal(*req.Params, &params); err != nil {
+			return nil, err
+		}
+		h.workspace.Store(params.TextDocument.URI, params.TextDocument.Text)
+		return nil, nil
+	case "textDocument/didChange":
+		var params lsp.DidChangeTextDocumentParams
+		if err := json.Unmarshal(*req.Params, &params); err != nil {
+			return nil, err
+		}
+
+		err := h.workspace.TrackEdit(params.TextDocument.URI, params.ContentChanges)
+		if err != nil {
```

Comment:
FYI in a case like this you could just do `return nil, h.workspace.TrackEdit(params.TextDocument.URI, params.ContentChanges)`